espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/w3c/models/InputSource.kt

```diff
@@ -161,50 +145,17 @@ class InputSource {
         private var y: Long? = null
         private var value: String? = null
         private var origin = Origin()
-        fun withType(type: ActionType?): ActionBuilder {
-            this.type = type
-            return this
-        }
-
-        fun withDuration(duration: Long?): ActionBuilder {
-            this.duration = duration
-            return this
-        }
-
-        fun withButton(button: Int?): ActionBuilder {
-            this.button = button
-            return this
-        }
-
-        fun withX(x: Long?): ActionBuilder {
-            this.x = x
-            return this
-        }
-
-        fun withY(y: Long?): ActionBuilder {
-            this.y = y
-            return this
-        }
-
-        fun withValue(value: String?): ActionBuilder {
-            this.value = value
-            return this
-        }
-
-        fun withOrigin(origin: Origin): ActionBuilder {
-            this.origin = origin
-            return this
-        }
-
-        fun withOrigin(originType: String?): ActionBuilder {
-            origin.type = originType
-            return this
-        }
-
-        fun withElementId(elementId: String?): ActionBuilder {
+        fun withType(type: ActionType?): ActionBuilder = apply { this.type = type }
+        fun withDuration(duration: Long?): ActionBuilder = apply { this.duration = duration }
+        fun withButton(button: Int?): ActionBuilder = apply { this.button = button }
+        fun withX(x: Long?): ActionBuilder = apply { this.x = x }
+        fun withY(y: Long?): ActionBuilder = apply { this.y = y }
+        fun withValue(value: String?): ActionBuilder = apply { this.value = value }
+        fun withOrigin(origin: Origin): ActionBuilder = apply { this.origin = origin }
+        fun withOrigin(originType: String?): ActionBuilder = apply { origin.type = originType }
```

Comment:
I assume this method should be called withOriginType

Reply:
it also makes sense to create origin object if this method is called and is is null